docs: Update backward-compatibility with @provides info (#2837)
apollographql/federation: docs/source/federation-2/backward-compatibility.mdx

This `@requires` directive should _break_ composition because it depends on subfields of `ProductDimensions` that _don't exist_ (`length` and `depth`).

#### Invalid `@provides` directives

> A subgraph can mark an entity field with the [`@provides` directive](https://www.apollographql.com/docs/federation/federated-types/federated-directives/#provides) to indicate that it resolve entity fields normally marked as `@external` if they are instead resolved through this particular path:

Reviewer: I found this explanation a little difficult to follow. What does "it" refer to in, "that it resolve[s] entity fields normally..."? Could we callout whatever "it" is explicitly to clarify?

Reviewer: Is it something like this:?

```suggestion
A subgraph can annotate an entity field with the [`@provides` directive](/federation/federated-types/federated-directives/#provides) to indicate that the subgraph can resolve entity fields normally marked as `@external` on its own.
```

<p style="margin-bottom: 0">✅</p>

```graphql title="Subgraph A"
type Product @key(fields: "id") {
  id: ID!
  info: ProductInfo @external
}

type ProductInfo {
  name: String! @external
  inStock: Boolean! @external
}

type Query {
  outOfStockProducts: [Product!]! @provides(fields: "info { name }")
  discontinuedProducts: [Product!]!
}
```

> In this example, if we traverse through the `outOfStockProducts` Query, Subgraph A can resolve the `Product.info.name` field, otherwise any other path will result in another subgraph call.

Reviewer: Tried to clarify this by breaking up the sentence + rewording. Hopefully still accurate:

```suggestion
In the above example, Subgraph A can resolve the `Product.info.name` field  when accessed through the `outOfStockProducts` query. Any other path to `Product.info.name` results in an additional subgraph call.
```

> Federation 1 _incorrectly_ allows a `@provides` directive such as the following:

Reviewer: ```suggestion
Federation 1 _incorrectly_ allows `@provides` usage like the following:
```

<p style="margin-bottom: 0">❌</p>

```graphql title="Subgraph A"
type Product @key(fields: "id") {
  id: ID!
  info: ProductInfo @external
}

type ProductInfo {
  name: String! @external
  inStock: Boolean! @external
}

type Query {
  outOfStockProducts: [Product!]! @provides(fields: "info")
  discontinuedProducts: [Product!]!
}

> 

Reviewer: ```suggestion
```

```

> This `@provides` directives should _break_ composition here because it does not specify which subfields of `ProductInfo` that it can resolve.

Reviewer: ```suggestion
The above `@provides` directives usage should _break_ composition because it does not specify which subfields of `ProductInfo` it can resolve.
```
